x-pack/plugin/async-search/src/main/java/org/elasticsearch/xpack/search/MutableSearchResponse.java

```diff
@@ -103,18 +113,35 @@ synchronized void updatePartialResponse(
         this.totalHits = totalHits;
         this.reducedAggsSource = reducedAggs;
         this.reducePhase = reducePhase;
+        if (isFinalLocalReduce) {
+            if (localClusterSearchCompleted == false
+                && clusters.getTotal() > 0
+                && clusters.getTotal() > (clusters.getSuccessful() + clusters.getSkipped())) {
```

Comment:
can you help me understand this conditional? I don't remember why it's needed.

Reply:
In all cases (that I know of) except `ccsMinimizeRoundtrips`, the Clusters object is created in its final state even at the start of an operation. By final state, I mean `successful + skipped = total`.

Examples: 

1) CCS NOT doing minimize_roundtrips it is created at the beginning with say `Clusters{total: 3; successful: 2; skipped: 1}`, so you don't want to increment successful here. 

2) In addition, there are cases where Clusters is initialized to Clusters.EMPTY (not totally sure why), but without this protection that causes those test cases to fail because you get `Clusters{total: 0; successful: 1; skipped: 0}` 

So this sanity check ensures that `successful` is incremented only in scenarios where the Cluster object has been set up to be incrementally updated (which right now only the ccsMinimizeRoundtrips scenario does).

Reply:
I added additional commentary.